### ISSUE TITLE
Files with same names doesn't open. 

### ISSUE THREAD

**Issue by hiddenruins** [nvim-telescope/telescope.nvim#697]
### Description
Hello, I am having this weird issue when opening files. Telescope won't switch to another file with the same name. I was playing around to see how it occurs and these are few things I know:-

- File opened first should be the nested file
- File should not be on buffers list.
- After selecting other files it works as normal. 
- If I open the root file first and try to open the nested file it works.

**Recording**
https://user-images.githubusercontent.com/5569621/112799860-a270f680-908e-11eb-82a4-fae2bd17d12d.mp4

**Expected Behavior**
Files should be opened by telescope.

**Actual Behavior**
Telescope shows the same file instead of opening the other file.

### Details

<details><summary>Reproduce</summary>

![image](https://user-images.githubusercontent.com/5569621/112798431-9a17bc00-908c-11eb-881d-e07fc9b1d6f1.png)

1. `nvim`.
2. Telescope find_files and open `folder/file.log`
3. Now try to open `file.log`.

</details>

<details><summary>Environment</summary>

- NVIM v0.5.0-dev+1186-g8665a96b9
- Build type: Release
- LuaJIT 2.1.0-beta3
- Telescope a7fa604

</details>

<details><summary>Configuration</summary>
<p>

```lua
telescope.setup{
  defaults = {
    vimgrep_arguments = {
      'rg',
      '--color=never',
      '--no-heading',
      '--with-filename',
      '--line-number',
      '--column',
      '--smart-case'
    },
    file_sorter =  sorters.get_fzy_sorter, -- better sorter than default
    file_ignore_patterns = {"node_modules", "*.zip", "*.rar"},
    generic_sorter =  sorters.get_fzy_sorter, -- better sorter than default
    shorten_path = true,
    winblend = 0,
    width = 0.75,
    preview_cutoff = 120,
    results_height = 1,
    results_width = 0.8,
    border = {},
    borderchars = { '─', '│', '─', '│', '╭', '╮', '╯', '╰'},
    color_devicons = true,
    use_less = true,
    set_env = { ['COLORTERM'] = 'truecolor' }, -- default = nil,
  }
}

```

**Comment by elianiva** [comment]
removing this line fixes the issue

https://github.com/nvim-telescope/telescope.nvim/blob/a7fa60411efbfddc0674d65346a5311e33f6deb1/lua/telescope/actions/set.lua#L111

though I'm not sure why we need this line

**Comment by Conni2461** [comment]
This line exist because otherwise we would do `:edit /home/conni/tmp/a/file` rather than `:edit a/file`.
The condition with `vim.fn.bufnr(filename)` is just wrong.

CC @oberblastmeister

**Comment by oberblastmeister** [comment]
What we can do is save the bufnr *before* we start the picker like we do here https://github.com/nvim-telescope/telescope.nvim/blob/a7fa60411efbfddc0674d65346a5311e33f6deb1/lua/telescope/pickers.lua#L317
Then we can check if the bufnr is the same without calling `vim.fn.bufnr`. This way we don't have to call `path.normalize` either.

**Comment by Conni2461** [comment]
We do not need to save the window upfront because we can just get it at that time. We also should not do `:edit` with the absolute path this will lead to absolute paths when doing `:buffers` and some statuslines. We had this issue in the beginning.
We should just move the normalize in the if block. Problem solved i think

**Comment by oberblastmeister** [comment]
What do you mean by
> We should just move the normalize in the if block.

**Comment by Conni2461** [comment]
```diff
diff --git a/lua/telescope/actions/set.lua b/lua/telescope/actions/set.lua
index ea46e83..08b3a9d 100644
--- a/lua/telescope/actions/set.lua
+++ b/lua/telescope/actions/set.lua
@@ -108,11 +108,10 @@ set.edit = function(prompt_bufnr, command)
   if entry_bufnr then
     edit_buffer(command, entry_bufnr)
   else
-    filename = path.normalize(vim.fn.fnameescape(filename), vim.loop.cwd())
-
     -- check if we didn't pick a different buffer
     -- prevents restarting lsp server
     if vim.api.nvim_get_current_buf() ~= vim.fn.bufnr(filename) then
+      filename = path.normalize(vim.fn.fnameescape(filename), vim.loop.cwd())
       vim.cmd(string.format("%s %s", command, filename))
     end
```

That works for me and it doesn't call `:edit` when the selected file is the current file

**Comment by oberblastmeister** [comment]
so `vim.fn.bufnr` works with absolute paths?l

**Comment by oberblastmeister** [comment]
i guess it works, vim help docs for `{expr}`
>                A full match is preferred, otherwise a match at the start, end
                or middle of the buffer name is accepted.  If you only want a
                full match then put "^" at the start and "$" at the end of the
                pattern.

**Comment by oberblastmeister** [comment]
> Otherwise maybe a str compare between nvim_buf_get_name(0) and filename would be enough.
 
I think that is the best solution. Then we can do normalize after.

**Comment by Conni2461** [comment]
Someone, who wants to fix it, can open a PR

**Comment by oberblastmeister** [comment]
Should be fixed with #741
